eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/IntegerConverter.php

```diff
@@ -121,20 +125,20 @@ public function getIndexColumn()
      *   - {@link self::HAS_MAX_VALUE}
      *   - {@link self::HAS_MIN_VALUE}.
      *
-     * @param int|null $minValue Minimum int value, or null if not set
-     * @param int|null $maxValue Maximum int value, or null if not set
+     * @param int|null|false $minValue Minimum int value, or null if not set
+     * @param int|null|false $maxValue Maximum int value, or null if not set
      *
      * @return int
      */
     private function getStorageDefValidatorState($minValue, $maxValue)
     {
         $state = 0;
 
-        if ($minValue !== null) {
+        if (!empty($minValue)) {
```

Comment:
`empty(0)` returns true, but 0 is a valid value. So it should rather use something like `is_int` or `is_numeric`, depending on whether the input may be a numeric string.